server/src/application/application.service.ts

```diff
@@ -21,6 +21,14 @@ export class ApplicationService {
     return;
   }
 
+  async getInProgressApplication(userId): Promise<Application> {
+    const application = await this.applicationRepository.findOne({
+      where: { user: userId, isSubmitted: false },
+    });
+
+    return application ? application : this.createApplication();
```

Comment:
This function shouldn't create an application. Maybe the controller can call or else rename the fn name to getOrCreateApplication